README.md

```diff
@@ -37,63 +32,237 @@ Let's look at each of the components in more detail.
+The schema is initially validated (see [validate](#validate) below), and will raise an exception if not a valid JSON Table Schema.  
 
-Model requires Promise to work properly, and need to be sure that Promise available globally. You are free to choose any Promise polyfill  
+```javascript
+var Schema = require('jsontableschema').schema;
+```
 
 ```javascript
-const model = new Schema('http://someurl.com/remote.json')
+var model = new Schema('http://someurl.com/remote.json')
 ```
 or
 ```javascript
-const model = new Schema({JSON OBJECT})
+var model = new Schema({JSON OBJECT})
 ```
-instance always returns promise
+instance always returns `Promise`
 ```javascript
-model.then(schema => {
+model.then(function(schema) {
     // working code to use schema model
-}).catch(error => {
+    var fields = schema.fields();
+   
+}).catch(function(error) {
     // something went wrong and error variable has explanations
 })
 ```
 
-Some methods available to Schema instances:
+Following methods are available on `Schema` instances:
 
-* `cast(fieldName, value, index, skipConstraints)` - returns a value cast against a named `fieldName`
-* `convertRow(...args)` - convert the arguments given to the types of the current schema <sup>1</sup>
-* `convert(items, failFast = false)` - convert an array of rows using the current schema of the Schema instance <sup>2</sup>
+* `castValue(fieldName, value, index, skipConstraints)` - returns a value cast against a named `fieldName`
+* `testValue(fieldName, value, index, skipConstraints)` - returns boolean after a check if value can be casted against a named `fieldName`
+* `castRow(items, failFast = false, skipConstraints = false)` - convert the arguments given to the types of the current schema <sup>1</sup>
 * `fields` - returns an array of the schema's fields
 * `foreignKeys` - returns the foreign key property for the schema
-* `getConstraints(fieldName, index = 0)` - return the constraints object for a given `fieldName` <sup>3</sup>
-* `getField(fieldName, index = 0)` - return the field object for `fieldName` <sup>3</sup>
+* `getConstraints(fieldName, index = 0)` - return the constraints object for a given `fieldName` <sup>2</sup>
+* `getField(fieldName, index = 0)` - return the field object for `fieldName` <sup>2</sup>
 * `getFieldsByType(typeName)` - return all fields that match the given type
+* `requiredHeaders` - returns headers with the `required` constraint as an array
 * `hasField(fieldName)` - checks if the field exists in the schema. Returns a boolean
 * `headers` - returns an array of the schema headers
-* `primaryKey` - returns the primary key field for the schema
-* `requiredHeaders` - returns headers with the `required` constraint as an array
-* `test(fieldName, value, index, skipConstraints)` - returns boolean after a check if value can be casted against a named `fieldName`
+* `primaryKey` - returns the primary key field for the schema  
```

Comment:
Again I'm voting for `primaryKey` property which always is **array** unlike spec array or string. Or any client should check it on every usage (https://github.com/frictionlessdata/jsontableschema-py/issues/60).